RealmCrustTests/CollectionMappingTests.swift

```diff
@@ -15,13 +15,134 @@ class CollectionMappingTests: RealmMappingTest {
+        try! self.adaptor!.save(objects: [ original ])
+        XCTAssertEqual(Company.allObjects(in: realm!).count, 1)
+        XCTAssertEqual(Employee.allObjects(in: realm!).count, 1)
+        
+        let companyStub = CompanyStub()
+        let employeeStub = EmployeeStub()
+        let employeeStub2 = EmployeeStub()
+        companyStub.employees = [employeeStub, employeeStub, employeeStub2, employeeStub]
+        let json = try! JSONValue(object: companyStub.generateJsonObject())
+        
+        let mapping = CompanyMappingReplaceUnique(adaptor: self.adaptor!)
+        let mapper = Mapper<CompanyMappingReplaceUnique>()
+        
+        let spec = Spec.mapping("", mapping)
+        let company = try! mapper.map(from: json, using: spec)
+        let employees = company.employees
+        
+        XCTAssertEqual(Employee.allObjects(in: realm!).count, 3)
+        XCTAssertEqual(employees.count, 2)
+        XCTAssertTrue(employeeStub.matches(object: employees[0]))
+        XCTAssertTrue(employeeStub2.matches(object: employees[1]))
+    }
+    
+    func testMappingCollectionByReplaceDeleteUnique() {
+        class CompanyMappingReplaceDeleteUnique: CompanyMapping {
+            override func mapping(tomap: inout Company, context: MappingContext) {
+                let employeeMapping = EmployeeMapping(adaptor: self.adaptor)
+                tomap.employees             <- (Spec.collectionMapping("employees", employeeMapping, (.replace(delete: { $0 }), true)), context)
+                tomap.uuid                  <- ("data.uuid" as JSONKeypath, context)
+            }
+        }
+        
+        let uuid = NSUUID().uuidString
+        
+        let original = Company()
+        let originalEmployee = Employee()
+        original.uuid = uuid
+        originalEmployee.uuid = uuid
+        original.employees.append(originalEmployee)
+        try! self.adaptor!.save(objects: [ original ])
+        XCTAssertEqual(Company.allObjects(in: realm!).count, 1)
+        XCTAssertEqual(Employee.allObjects(in: realm!).count, 1)
+        
+        let companyStub = CompanyStub()
+        let employeeStub = EmployeeStub()
+        let employeeStub2 = EmployeeStub()
+        companyStub.employees = [employeeStub, employeeStub, employeeStub2, employeeStub]
+        let json = try! JSONValue(object: companyStub.generateJsonObject())
+        
+        let mapping = CompanyMappingReplaceDeleteUnique(adaptor: self.adaptor!)
+        let mapper = Mapper<CompanyMappingReplaceDeleteUnique>()
+        
+        let spec = Spec.mapping("", mapping)
+        let company = try! mapper.map(from: json, using: spec)
+        let employees = company.employees
+        
+        // 3 because `founder`.
+        XCTAssertEqual(Employee.allObjects(in: realm!).count, 2)
+        XCTAssertEqual(employees.count, 2)
+        XCTAssertTrue(employeeStub.matches(object: employees[0]))
+        XCTAssertTrue(employeeStub2.matches(object: employees[1]))
+    }
```

Comment:
These new tests don't actually work, because I first have to do my next major task of overhauling how primary keys are mapped, but figured I should have tests anyway :).